nucleus/common/common-util/src/main/java/com/sun/common/util/logging/GFLogRecord.java

```diff
@@ -84,4 +84,31 @@ public String getThreadName() {
     public void setThreadName(String threadName) {
         this.threadName = threadName;
     }
+
+    /**
+     * CUSTOM-55
+     * in case of an object passed as a parameter, call it's toString() method
+     * to resolve it's values in the current thread, instead of waiting for queues / etc
+     * so there is no possibility of state change of the object between threads
+     * Append the original parameters at the end, as they are used for by some logging formatters,
+     * such as JSON logging formatter for context
+     *
+     * @param params
+     * @return parameter array
+     */
+    private Object[] transformParameters(Object[] params) {
+        if (params == null) {
+            return null;
+        }
+        Object[] result = new Object[params.length * 2];
+        for (int stringIndex = 0, originalIndex = params.length; stringIndex < params.length;
```

Comment:
I would probably use paramsIndex instead of originalIndex as a name

Reply:
Performance is unaffected because toString() gets called eventually anyway.
`JsonLogFormatIT` works because now parameters include both strings and original objects.
All that's is original objects so memory consumption has only tiny increase at the end of the day, just due to object array being bigger

Reply:
> Also if we would use both SysLogHandler and FileLoggerHandler, GFLogRecord would be created in bot

fixed